README.md
Repo for the parent and child themes on my personal website.


hestia/CHANGELOG.md

 ### v1.1.77 - 2018-05-24 
 **Changes:** 
 * Fixed compatibility with WP GDPR Compliance plugin
* Fixed problem with the Big Title Background not being able to be removed
* Fixed problem with default content added in the Very Top Bar section
 
 ### v1.1.76 - 2018-05-18 
 **Changes:** 
 * Fixed problem with some redundant strings in the lite version
 
 ### v1.1.75 - 2018-05-18 
 **Changes:** 
 * NEW Feature to allow a widget area in the Big Title/Slider section
* [Hestia PRO] Fixed alignment for the Authors Section on the Blog page
* Fix fatal error on lower PHP versions
 
 ### v1.1.74 - 2018-05-10 
 **Changes:** 
 * [Hestia PRO] New Icon picker option in the Pricing section
* Fixed problem with title meta in the Archive page header when the language is different than English
* Fixed problem with duplicated titles when using the second header layout option
* Fixed problem with the Big Title Second Button control not working in the Orfeo child theme
* Fixed some layout issues with the WooCommerce filtering widgets in the Footer area
 
 ### v1.1.73 - 2018-04-24 
 **Changes:** 
 * Improved Cart page layout on mobile
* Fixed problem with the About and Contact sections editor not saving the input
* Fixed problem with footer section on internal pages in IE11
* Removed meta on the Archive pages - Doc to add them back https://docs.themeisle.com/article/793-how-to-add-back-category-tag-author-in-the-archive-title-in-hestia
 
 ### v1.1.72 - 2018-04-17 
 **Changes:** 
 * Compatibility with Elementor Pro 2.0 theme location API header and footer
* NEW extra options for the Big title/Slider area - Parallax effect 
* UX Customize layout improvements
* Fixed layout compatibility issues with the sidebar on bbpress and BuddyPress pages
* Fixed layout of the Search products widget
* Allow a shortcode for the Contact section
* Improved compatibility with Pirate Forms Extended plugin
* Fixed issue with sidebar position when the option for logged in users to be able to post comments is enabled
* Fixed problem with footer sticking to the bottom of the page when to little content
* Made reCaptcha button smaller on mobile devices to better accommodate the layout
 
 ### v1.1.71 - 2018-03-14 
 **Changes:** 
 * Fixed problem with the preview images not showing anymore on the blog page
* Fixed problem with the parallax effect on inner pages
 
 ### v1.1.70 - 2018-03-08 
 **Changes:** 
 * Fixed alignment issues with the Featured posts area
 
 ### v1.1.69 - 2018-03-08 
 **Changes:** 
 * Fixed alignment issue in the featured posts area
 
 ### v1.1.68 - 2018-03-07 
 **Changes:** 
 * Fixed compatibility issues with WooCommerce 3.3 thumbnail cropping options
 
 ### v1.1.67 - 2018-03-06 
 **Changes:** 
 * NEW Featured posts area section on the blog
* Fixed compatibility issues with WooCommerce 3.3 thumbnail cropping options
* Fixed compatibility issues with WeGlot plugin language switcher
* Fixed problems with synchronization between the About section and the front page
* Fixed compatibility with the Invisible reCaptcha plugin
* [Hestia PRO] Removed constricted video dimensions
* Made Social Sharing strings translatable
* Fixed fatal error with PHP 5.2
* Fixed issue with random images appearing on the blog page in child themes
 
 ### v1.1.66 - 2018-02-19 
 **Changes:** 
 * Fixed responsive alignment on the shop page
 
 ### v1.1.65 - 2018-02-16 
 **Changes:**
 * Fixed compatibility issues with Pirate Forms on mobile
* Fixed warning appearing in the Blog section
* [Hestia PRO] Fixed empty cart layout in navigation area
* Fixed issues with the Header layout control
 
 ### v1.1.63 - 2018-02-14 
 **Changes:** 
 * [Hestia PRO] Option to open portfolios in a lightbox
* [Hestia PRO] Added animations for frontpage sections
* [Hestia PRO] Fixed compatibility issues for the navigation cart with variable products
* [Hestia PRO] Improved navigation cart layout on mobile devices
* [Hestia PRO] Allowed some formatting tags in the Slider section buttons
* New option for choosing the Header area layout
* [Hestia PRO] Two new layouts for the Header area
* Small fixes for WooCommerce widgets and product layouts
* Made entire team member boxes clickable
* Fixed problem with third level menu items getting hidden when parent menu item had description
* Fixed alignment problem on IE11 for megamenu
* Fixed problem with dropdown menu items on Firefox
* Added styling for more widgets in the Subscribe section
* Removed characters limitation for WooCommerce product's titles
 
 ### v1.1.62 - 2018-01-31 
 **Changes:** 
 * Improved compatibility with WooCommerce 3.3
* Fixed problem with container width control not working properly
* Added missing about-contact.png image
* Fixed warning: Implode function invalid arguments passed
 
 ### v1.1.61 - 2018-01-30 
 **Changes:** 
 * Fixed problem with dismissable notification
 
 ### v1.1.60 - 2018-01-29 
 **Changes:** 
 * [Hestia PRO] New option to disable the transparent state of the navigation bar
* [Hestia PRO] New option for changing the color in the hover state of the navigation bar items
* [Hestia PRO] New Category select option for the Blog frontpage section
* [Hestia PRO] Fix video header overflowing body
* New Font size controls for Frontpage sections
* Improved migration function from the old font size controls
* Added hooks for the Clients Bar and Ribbon sections
* Fixed problems with navigation color options for drop-down menus
* Basic enhancements to the editor area
* Small layout fixes for WooCommerce pages
* Fixed synchronization function for the About frontpage section image and content
* Optimize theme for a smaller size archive
* Improved compatibility with Pirate Forms plugin
* Made footer stick to the bottom of the pages no matter the content size
* Customizer selective refresh UX changes

hestia-child/js/zonesmap.js
/*
* Author: Connor Riley
* Zoning map of Richmond, VA
* Version: 1.0
* www.connorriley.me
* Made with Google Maps APIs
* 
*/

//var globalMap;
var places = [];
var richmond = {lat: 37.551441, lng: -77.436056};

function start() {
    initMap('zone-map');
    //createBlankMap('retail-map');
}

function createBlankMap(blankMap) {

    var customMapType = new google.maps.StyledMapType(
        [
            {
                elementType: 'labels',
                stylers: [{visibility: 'off'}]
            }
        ],
        {
            name: 'Custom Style'
        }
    );

    var map = new google.maps.Map(document.getElementById(blankMap), {
        center: richmond,
        zoom: 13
    });
    map.mapTypes.set(customMapTypeId, customMapType);
    map.setMapTypeId(customMapTypeId);
}

function initMap(mapID) {

    //var layer = 'toner-lite';

    var map = new google.maps.Map(document.getElementById(mapID), {
        center: richmond,
        zoom: 12
    });

    //globalMap = map;

    customMapTypeId = 'rvazoningmap';

    /*try {
        map.mapTypes.set(layer, new google.maps.StamenMapType(layer));
    } catch(e) {
        console.log(e);
        map.mapTypes.set(layer, new google.maps.StyledMapType());
    }*/
    map.mapTypes.set(customMapTypeId, new google.maps.StyledMapType());
    map.setMapTypeId(customMapTypeId);

    //getShapes(map, 'CARE');
    //getShapes(map, 'Enterprise Zones');
    getHistoric(map);

    map.data.setStyle(function(feature) {
        var id = feature.getId();
        console.log(id);
        var color = id == 'III' || id == '1' ? '#6e9489' : '#FF3800';
        return {
            fillColor: color,
            strokeOpacity: 0.8,
            strokeWeight: 1,
            strokeColor: color,
            fillOpacity: 0.5
        };
    });

}

function getShapes(map, str) {
    jQuery.getJSON('inc/'+str+'.geojson', function(data) {
        map.data.addGeoJson(data, {idPropertyName: 'name'});
    });
}

function getHistoric(map) {
    jQuery.getJSON('inc/rows.json', function( data ) {
        console.log(data[0]);
    });
}

/*
function getShapes(map, str, color) {
    color='red';
    jQuery.getJSON('inc/'+str+'.geojson', function( data ) {
        console.log(data);
        var items = [];
        jQuery.each( data.features, function( key, val ) {
            var geometry = val.geometry.coordinates[0][0];
            var name = val.properties.name;
            var coords = [];
            var count = 0;
            jQuery.each( geometry, function( key, val ) {
                var lat = parseFloat(val[1]);
                var lng = parseFloat(val[0]);
                var set = {lat: lat, lng: lng};
                coords.push(set);
                count++;
            });
            console.log(count);
            var poly = new google.maps.Polygon({
                paths: coords,
                strokeColor: color,
                strokeOpacity: 0.8,
                strokeWeight: 3,
                fillColor: color,
                fillOpacity: 0.35
            });
            poly.setMap(map);
        });

    });

}*/



jQuery(document).ready(function($) {

    start();
    //auto runs on click of default search buttons
    $('#developers-map-wrap input[type="radio"].auto-run').click(function(){
        {
            initMap('developers-map');
        }
    });

    $('#meta-map-wrap input[type="radio"].auto-run').click(function(){
        if ($(this).is(':checked'))
        {
            initMap('meta-map');
            $('button[name=meta-run-button]').fadeOut( "slow" );
        }
    });


    //Activate Map buttons
    $('button[name=run-button]').click(function(){
        initCustomMap();
    });

});


hestia-child/js/rvaheatmap.js
/*
* Author: Connor Riley
* Heat Maps of Richmond, VA
* Version: 1.0
* www.connorriley.me
* Made with Google Maps APIs
*
*/
var globalMap;
var heatmap;
var infowindow;
var service;
var richmond = {lat: 37.551441, lng: -77.436056};

function start() {
    initMap('developers-map');
    createBlankMap('retail-map');
    createBlankMap('office-map');
    createBlankMap('public-map');
    createBlankMap('todo-map');
    createBlankMap('meta-map');
    createBlankMap('custom-map');
}

function createBlankMap(blankMap) {

    var customMapType = new google.maps.StyledMapType(
        [
            {
                elementType: 'labels',
                stylers: [{visibility: 'off'}]
            }
        ],
        {
            name: 'Custom Style'
        }
    );

    var map = new google.maps.Map(document.getElementById(blankMap), {
        center: richmond,
        zoom: 13
    });
    map.mapTypes.set(customMapTypeId, customMapType);
    map.setMapTypeId(customMapTypeId);
}

function initMap(mapID) {

    var customMapType = new google.maps.StyledMapType(
        [
            {
                elementType: 'labels',
                stylers: [{visibility: 'on'}]
            }
        ],
        {
            name: 'Custom Style'
        }
    );

    var map = new google.maps.Map(document.getElementById(mapID), {
        center: richmond,
        zoom: 13
    });

    globalMap = map;

    infowindow = new google.maps.InfoWindow({maxWidth: 200});

    customMapTypeId = 'rvaheatmap';
    var selector = typeSelector(mapID);
    var types = selector[0];
    var keyword = selector[1];

    service =new google.maps.places.PlacesService(map);
    if(keyword != '') {
        service.radarSearch({
            location: richmond,
            radius: 5000,
            keyword: keyword
        }, callback);
    } else {
        service.radarSearch({
            location: richmond,
            radius: 5000,
            types: types
        }, callback);
    }

    var showBikeLanes = jQuery('#todo-bikelanes-select').prop("checked");

    if(mapID === 'todo-map' && showBikeLanes) {
        var bikeLayer = new google.maps.BicyclingLayer();
        bikeLayer.setMap(map);
    }
    map.mapTypes.set(customMapTypeId, customMapType);
    map.setMapTypeId(customMapTypeId);

}

function initCustomMap() {
    var richmond = {lat: 37.551441, lng: -77.436056};

    var customMapType = new google.maps.StyledMapType(
        [
            {
                elementType: 'labels',
                stylers: [{visibility: 'on'}]
            }
        ],
        {
            name: 'Custom Style'
        }
    );

    var map = new google.maps.Map(document.getElementById('custom-map'), {
        center: richmond,
        zoom: 13
    });

    globalMap = map;

    infowindow = new google.maps.InfoWindow({maxWidth: 200});

    customMapTypeId = 'rvaheatmap';

    var keyword = jQuery('#user-keyword').val();


    service = new google.maps.places.PlacesService(map);
    service.radarSearch({
        location: richmond,
        radius: 5000,
        keyword: keyword
    }, callback);

    map.mapTypes.set(customMapTypeId, customMapType);
    map.setMapTypeId(customMapTypeId);

}

function typeSelector(mapID) {
    var searchString = '#'+mapID+'-wrap input[name='+mapID+'-layer-scheme]:checked';
    var selected = jQuery(searchString).val();
    switch(selected){
            //Developers
        case 'booze':
            return [['bar', 'liquor_store', 'night_club'],''];
        case 'coffee':
            return ['','coffee'];
        case 'grocery':
            return [['grocery_or_supermarket'], ''];
        case 'gym':
            return [['gym'], ''];
        case 'restaurant':
            return [['restaurant', 'cafe', 'bakery', 'meal_takeaway'], ''];
        case 'parking':
            return [['parking'], ''];
            ///Retail
        case 'retail-all':
            return [['bakery', 'restaurant', 'cafe', 'food', 'grocery_or_supermarket', 'meal_delivery', 'meal_takeaway',
                     'beauty_salon', 'bicycle_store', 'book_store', 'car_dealer', 'florist', 'hair_care', 'movie_rental',
                     'pharmacy', 'shopping_mall', 'store', 'clothing_store', 'convenience_store', 'department_store',
                     'electronics_store', 'furniture_store', 'hardware_store', 'home_goods_store', 'jewelry_store',
                     'pet_store', 'shoe_store', 'liquor_store', 'beauty_salon', 'car_rental', 'hair_care', 'movie_rental',
                     'pharmacy', 'spa'],''];
        case 'retail-food':
            return [['bakery', 'restaurant', 'cafe', 'grocery_or_supermarket', 'meal_delivery', 'meal_takeaway', 'food'],''];
        case 'retail-goods':
            return [['beauty_salon', 'bicycle_store', 'book_store', 'car_dealer', 'florist', 'hair_care', 'movie_rental',
                     'pharmacy', 'shopping_mall', 'clothing_store', 'department_store',
                     'electronics_store', 'furniture_store', 'hardware_store', 'home_goods_store', 'jewelry_store',
                     'pet_store', 'shoe_store', 'liquor_store', 'convenience_store', 'store'], ''];
        case 'retail-service':
            return [['beauty_salon', 'car_rental', 'hair_care', 'movie_rental', 'spa', 'pharmacy' ], ''];
            //Office
        case 'office-other':
            return [['accounting', 'dentist', 'finance', 'physiotherapist', 'real_estate_agency', 'travel_agency'],''];
        case 'office-finance':
            return ['','finance'];
        case 'office-marketing':
            return ['', 'marketing'];
        case 'office-advertising':
            return ['', 'advertising'];
        case 'office-architect':
            return ['', 'architect'];
        case 'office-web':
            return ['','web'];
        case 'office-doctor':
            return [['doctor'],''];
        case 'office-lawyer':
            return [['lawyer'], ''];
            //Public
        case 'public-all':
            return [['school', 'city_hall', 'courthouse', 'embassy', 'fire_station', 'local_government_office', 'police',
                     'cemetery', 'church', 'hindu_temple', 'mosque', 'place_of_worship', 'synagogue', 'dentist', 'doctor',
                     'hospital'],''];
        case 'public-schools':
            return [['school'],''];
        case 'public-government':
            return [['city_hall', 'courthouse', 'embassy', 'fire_station', 'local_government_office', 'police'], ''];
        case 'public-religious':
            return [['cemetery', 'church', 'hindu_temple', 'mosque', 'synagogue', 'place_of_worship'], ''];
        case 'public-medical':
            return [['dentist', 'hospital', 'doctor'], ''];
            //Todo
        case 'todo-all':
            return [['amusement_park', 'aquarium', 'art_gallery', 'bowling_alley', 'casino', 'night_club', 'museum', 'zoo',
                     'gym', 'bicycle_store', 'health', 'stadium', 'campground', 'park', 'rv_park'],''];
        case 'todo-amusement':
            return [['amusement_park', 'aquarium', 'bowling_alley', 'casino', 'night_club', 'museum', 'zoo',
                     'stadium', 'art_gallery'],''];
        case 'todo-fitness':
            return [['gym', 'bicycle_store', 'stadium'], ''];
        case 'todo-greenspace':
            return [['campground', 'park', 'rv_park'], ''];
            //meta
        case 'meta-office':
            return [['accounting', 'dentist', 'finance', 'physiotherapist', 'real_estate_agency',
                     'travel_agency', 'doctor', 'lawyer'],''];
        case 'meta-retail':
            return [['bakery', 'restaurant', 'cafe',  'grocery_or_supermarket', 'meal_delivery', 'meal_takeaway',
                     'beauty_salon', 'bicycle_store', 'book_store', 'car_dealer', 'florist', 'hair_care', 'movie_rental',
                     'pharmacy', 'shopping_mall', 'spa', 'store', 'clothing_store', 'convenience_store', 'department_store',
                     'electronics_store', 'furniture_store', 'hardware_store', 'home_goods_store', 'jewelry_store',
                     'pet_store', 'shoe_store', 'liquor_store', 'beauty_salon', 'car_rental', 'hair_care', 'movie_rental',
                     'food', 'pharmacy'],''];
        case 'meta-public':
            return [['school', 'city_hall', 'courthouse', 'embassy', 'fire_station', 'local_government_office', 'police',
                     'cemetery', 'church', 'hindu_temple', 'mosque', 'synagogue', 'place_of_worship', 'hospital'],''];
            ///Custom
        case 'other':
            var kw = jQuery('#user-keyword').val();
            return ['', kw];
            //Default to none
        default:
            return [[''],''];
    }

}

function callback(results, status) {
    if (status === google.maps.places.PlacesServiceStatus.OK) {
        var places = [];
        for (var i = 0; i < results.length; i++) {
            places[i] = results[i].geometry.location;
        }
//        createHeatMap(map);
        generateHeatMap(places);
    }
}

//function createHeatMap(globalMap) {
//    heatmap = new google.maps.visualization.HeatmapLayer({
//        data: places,
//        map: map
//    });
//    console.log(places);
//    heatmap.set('radius',25);
//    places = [];
//}

function generateHeatMap(places) {
    heatmap = new google.maps.visualization.HeatmapLayer({
        data: places,
        map: globalMap
    });
    try {
        var oldData = heatmap.getData().j;
        var newData = oldData.concat(places);
    } catch(e) {
        var newData = places;
    }
    heatmap.setData(newData);
    heatmap.set('radius',25);
//    search(++searchNumG);
}

jQuery(document).ready(function($) {
    //auto runs on click of default search buttons
    $('#developers-map-wrap input[type="radio"].auto-run').click(function(){
        {
            initMap('developers-map');
        }
    });

    $('#retail-map-wrap input[type="radio"].auto-run').click(function(){
        if ($(this).is(':checked'))
        {
            initMap('retail-map');
            $('button[name=retail-run-button]').fadeOut( "slow" );
        }
    });

    $('#office-map-wrap input[type="radio"].auto-run').click(function(){
        if ($(this).is(':checked'))
        {
            initMap('office-map');
            $('button[name=office-run-button]').fadeOut( "slow" );
        }
    });

    $('#public-map-wrap input[type="radio"].auto-run').click(function(){
        if ($(this).is(':checked'))
        {
            initMap('public-map');
            $('button[name=public-run-button]').fadeOut( "slow" );
        }
    });

    $('#todo-map-wrap input[type="radio"].auto-run').click(function(){
        if ($(this).is(':checked'))
        {
            initMap('todo-map');
            $('button[name=todo-run-button]').fadeOut( "slow" );
        }
    });

    $('#meta-map-wrap input[type="radio"].auto-run').click(function(){
        if ($(this).is(':checked'))
        {
            initMap('meta-map');
            $('button[name=meta-run-button]').fadeOut( "slow" );
        }
    });


    //Activate Map buttons
    $('button[name=run-button]').click(function(){
        initCustomMap();
    });

    $('button[name=retail-run-button]').click(function(){
        initMap('retail-map');
        $(this).fadeOut( "slow" );
    });

    $('button[name=office-run-button]').click(function(){
        initMap('office-map');
        $(this).fadeOut( "slow" );
    });

    $('button[name=public-run-button]').click(function(){
        initMap('public-map');
        $(this).fadeOut( "slow" );
    });

    $('button[name=todo-run-button]').click(function(){
        initMap('todo-map');
        $(this).fadeOut( "slow" );
    });

    $('button[name=meta-run-button]').click(function(){
        initMap('meta-map');
        $(this).fadeOut( "slow" );
    });

    //Runs on bike lanes click
    $('#todo-bikelanes-select').click(function(){
        initMap('todo-map');
    });

    //runs custom search on enter key press
    $('#user-keyword').keypress(function (e) {
        var key = e.which;
        if(key == 13)  // the enter key code
        {
            initCustomMap();
        }
    });
    
    $('.panel-title a').click(function(){$(this).parent().delay(500).slideUp(1000)});
});


hestia-child/page-gallery.php
<?php
/**
* Template Name: Gallery
*/

get_header(); ?>

<section>
<?php if( have_posts()) : while( have_posts()) : the_post(); ?>
	<div class="da-row">
		   <h1><?php the_title(); ?></h1>

		  <p><?php the_content(); ?></p>
	</div>
<?php endwhile; endif; ?>
</section>

<section>
	<div class = "da-row group">
<?php
	$args = array ('post_type' => 'portfolio');
	$query = new WP_Query($args);

	if( $query->have_posts()) : while( $query->have_posts()) : $query->the_post(); ?>
		<div class="center table third-page">
			<h2><?php the_title(); ?></h2>
			<a href="<?php the_permalink(); ?>"><?php the_post_thumbnail('medium') ?></a>
		</div>

		<?php $count = $query->current_post + 1;
		if($count % 3 == 0) : ?>
			</div><div class = "da-row group">
		<?php endif; ?>

	<?php endwhile; endif; wp_reset_postdata();?>
	</div>
</section>

<?php get_footer(); ?>


hestia-child/single-gallery.php
<?php
/**
 * The template for displaying all single galleries and attachments.
 *
 * @package Hestia
 * @since Hestia 1.0
 */

get_header();

/**
 * Don't display page header if header layout is set as classic blog.
 */
$hestia_header_layout = get_theme_mod( 'hestia_header_layout', 'default' );
if ( $hestia_header_layout !== 'classic-blog' ) {
	hestia_display_page_header( 'post' );
}
?>

</header>

<div class="<?php echo hestia_layout(); ?>">
	<div class="blog-post blog-post-wrapper">
			<div class="container">

				<?php
				if ( have_posts() ) :
					while ( have_posts() ) :
						the_post();
						get_template_part( 'template-parts/content', 'single' );
					endwhile;
				else :
					get_template_part( 'template-parts/content', 'none' );
				endif;
				?>
				<?php $images = get_field('gallery');

if( $images ): ?>
    <ul>
        <?php foreach( $images as $image ): ?>
            <li>
                <a href="<?php echo $image['url']; ?>">
                     <img src="<?php echo $image['sizes']['thumbnail']; ?>" alt="<?php echo $image['alt']; ?>" />
                </a>
                <p><?php echo $image['caption']; ?></p>
            </li>
        <?php endforeach; ?>
    </ul>
<?php endif; ?>
			</div>
	</div>
</div>


<div class="footer-wrapper">
	<?php get_footer(); ?>


hestia-child/js/googlemap.js
var map;
var infowindow;

function initMap() {
    var carytown = {lat: 37.554123, lng: -77.483573}

    var customMapType = new google.maps.StyledMapType(
        [
            {
                elementType: 'labels',
                stylers: [{visibility: 'off'}]
            }
        ],
        {
            name: 'Custom Style'
        }
    );

    map = new google.maps.Map(document.getElementById('map'), {
        center: carytown,
        zoom: 16
    });

    infowindow = new google.maps.InfoWindow({maxWidth: 200});

    customMapTypeId = 'carrytown';

    service = new google.maps.places.PlacesService(map);
    service.radarSearch({
        location: carytown,
        radius: 500,
        types: ['store', 'shoe_store', 'restaurant', 'cafe', 'grocery_or_supermarket', 'clothing_store', 'point_of_interest', 'book_store', 'establishment', 'health']
    }, callback);

    map.mapTypes.set(customMapTypeId, customMapType);
    map.setMapTypeId(customMapTypeId);
}

function callback(results, status) {
    if (status === google.maps.places.PlacesServiceStatus.OK) {
        for (var i = 0; i < results.length; i++) {
            createLabel(results[i]);
        }
    }
}

function createLabel(place) {

    var request = {
        reference : place.reference,
    };

    service = new google.maps.places.PlacesService(map);

    service.getDetails(request, function(details, status) {
        if (status === google.maps.places.PlacesServiceStatus.OK) {
            var content_list = "<p style='color: #000'>" + details.name + ": </p><ul style='color: #000'>";
            for (var j = 0; j < details.types.length; j++) {
                content_list += '<li>' + details.types[j] + "</li>";
            }
            content_list += '</ul>';
            createMarker(details, content_list);
        } else if (status === google.maps.GeocoderStatus.OVER_QUERY_LIMIT) {
            setTimeout(function() {
                createLabel(place);
            }, 200);
        }
    });
}

function createMarker(place, content_list) {
    var placeLoc = place.geometry.location;

    var resultColor = colorPicker(place.types[0]);

    var marker = new google.maps.Marker({
        map: map,
        position: place.geometry.location,
        opacity: 0.75,
        icon: {
            path: google.maps.SymbolPath.CIRCLE,
            fillColor: resultColor,
            fillOpacity: .5,
            strokeColor: 'white',
            strokeWeight: .5,
            scale: 12
        }
    });

    google.maps.event.addListener(marker, 'mouseover', function() {
        marker.setOpacity(1);
        marker.setIcon({
            path: google.maps.SymbolPath.CIRCLE,
            fillColor: resultColor,
            fillOpacity: 1,
            strokeColor: 'white',
            strokeWeight: .5,
            scale: 14
        });
    });

    google.maps.event.addListener(marker, 'mouseout', function() {
        marker.setOpacity(.75);
        marker.setIcon({
            path: google.maps.SymbolPath.CIRCLE,
            fillColor: resultColor,
            fillOpacity: .5,
            strokeColor: 'white',
            strokeWeight: .5,
            scale: 12
        });
    });

    google.maps.event.addListener(marker, 'click', function() {
        infowindow.setContent(content_list);
        infowindow.open(map, this);
    });
}

function colorPicker(type) {
    switch (type) {
        case 'store':
            return 'blue';
        case 'clothing_store':
            return 'blue';
        case 'shoe_store':
            return 'blue';
        case 'bar':
            return 'green';
        case 'book_store':
            return 'yellow';
        case 'restaurant':
            return 'orange';
        case 'cafe':
            return 'orange';
        case 'meal_delivery':
            return 'orange';
        case 'grocery_or_supermarket':
            return 'purple';
        default:
            return 'grey';
    }
}
